DependencyInjection/SonataMediaExtension.php

```diff
@@ -44,6 +44,27 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('extra.xml');
         $loader->load('form.xml');
         $loader->load('gaufrette.xml');
+
+        $defS3 = $container->getDefinition('sonata.media.adapter.service.s3');
+        if (method_exists($defS3, 'setFactory')) {
+            // NEXT_MAJOR: Inline in gaufrette.xml
```

Comment:
please add the line to the gaufrette.xml and uncomment this and add a comment, to uncomment this in the next major, in this case we don't need the comment here 👍